pkg/api/v1/clients/factory/resource_client_factory.go

```diff
@@ -81,11 +99,15 @@ type ResourceClientFactory interface {
 // If SkipCrdCreation is set to 'true', the clients built with this factory will not attempt to create the given CRD
 // during registration. This allows us to create and register resource clients in cases where the given configuration
 // contains a token associated with a user that is not authorized to create CRDs.
+// Clients built with this factory will be able to access only resources the given namespace list. If no value is provided,
+// clients will be able to access resources in all namespaces.
 type KubeResourceClientFactory struct {
 	Crd             crd.Crd
 	Cfg             *rest.Config
 	SharedCache     *kube.KubeCache
 	SkipCrdCreation bool
+	Namespaces      []string
```

Comment:
suggestion: replace `Namespaces` with something like `NamespaceWhitelist` so it's more clear what function the list serves

Reply:
Good point, will do